### ISSUE TITLE
TbSync/EAS for Thunderbird 102

### ISSUE THREAD

**Issue by InFerYes** [jobisoft/EAS-4-TbSync#167]
I have released the final versions of TbSync and of the DAV provider in their 4.x branches:
https://github.com/jobisoft/TbSync/releases/
https://github.com/jobisoft/DAV-4-TbSync/releases/

This is a pre-release of the EAS provider:
https://github.com/jobisoft/EAS-4-TbSync/releases/

More information about the state of the EAS pre-release can be found here:
https://github.com/jobisoft/EAS-4-TbSync/issues/167#issuecomment-1252806380

**TbSync/EAS is NOT ready for productive use!**

<hr>

**NOTE: To avoid data loss, I will release TbSync for Thunderbird 102 only after https://bugzilla.mozilla.org/show_bug.cgi?id=1792399 has been backported to Thunderbird 102.**

<hr>

**Comment by jobisoft** [comment]
ETA : 2 weeks

**Comment by fridrich** [comment]
How far is this?

**Comment by domibln** [comment]
Please don't annoy him. You do want him to spend his spare time on fixing EAS-4-TbSync, not on answering posts, don't you?

**Comment by jobisoft** [comment]
I am on it, I use the provider myself, so there will be an update. I still need time.

**Comment by black-snow** [comment]
Looking forward to it and thanks for this handy addon :)

**Comment by ggruber** [comment]
would be really helpful to get a 102 compatible version

**Comment by jobisoft** [comment]
Here is a very preliminary version for 102. Please do NOT use it on your production profile.

Create a new profile and test it - and assume that it will eat all your data, so have some sort of backup of your calendar and contacts data.

The address book UI has changed massively and I have not adapted it yet. This version aims for the following:
* does calendar sync work on a new profile (up and down)
* for the brave ones: does calendar sync work on an updated profile, using an existing sync state?

[TbSync-beta.zip](https://github.com/jobisoft/EAS-4-TbSync/files/9581760/TbSync-beta.zip)
[EAS-4-TbSync.zip](https://github.com/jobisoft/EAS-4-TbSync/files/9581763/EAS-4-TbSync.zip)

**Comment by kxxt** [comment]
Everything is fine with my old outlook profile.

![Screenshot_87](https://user-images.githubusercontent.com/18085551/190586075-b0195124-ece6-471c-81d7-d310cf99211f.png)

**Comment by jobisoft** [comment]
A very big thanks to @kxxt for helping me with the update. Do you happen to have a paypal account, so I can forward the donations I received for EAS?

**Comment by kxxt** [comment]
> A very big thanks to @kxxt for helping me with the update. Do you happen to have a paypal account, so I can forward the donations I received for EAS?

I do have a paypal account(rsworktech@outlook.com) but feel free to keep the donations to yourself because you are the one who is paying a lot of attention to maintain TbSync.

**Comment by rubennc91** [comment]
It works fine for me, with Windows 10 and TB 102.2.2 (64 bit). 

Thank you very much @jobisoft!

**Comment by jobisoft** [comment]
@kxxt:

> I do have a paypal account([rsworktech@outlook.com](mailto:rsworktech@outlook.com)) but feel free to keep the donations to yourself because you are the one who is paying a lot of attention to maintain TbSync.

You should have received something :-) Thanks for digging through the code.

**Comment by kxxt** [comment]
> @kxxt:
> 
> > I do have a paypal account([rsworktech@outlook.com](mailto:rsworktech@outlook.com)) but feel free to keep the donations to yourself because you are the one who is paying a lot of attention to maintain TbSync.
> 
> You should have received something :-) Thanks for digging through the code.

Thanks for the donation. Actually I think it's too much for me. 

Also I learned how a Thunderbird plugin works and how to debug it(I have no experience before).

**Comment by lguerard** [comment]
New version managed to synchronise calendar, thanks a lot !

With the old TB I was able to get all the email address from the server when typing a new message but this is not showing anymore, is this normal ?

Thanks both for your hard work ! :)

**Comment by jobisoft** [comment]
> With the old TB I was able to get all the email address from the server when typing a new message but this is not showing anymore, is this normal ?

This update is not including an update for contact sync/handling yet.

**Comment by lguerard** [comment]
That makes sense, thanks a lot for the info :). I'll keep an eye out for this then !

**Comment by stheine** [comment]
> Here is a very preliminary version for 102. Please do NOT use it on your production profile.

I can confirm this to work ok.

**Comment by Trapulo** [comment]
Also ok for me (calendars, not contacts)

thanks!

**Comment by wolffi** [comment]
calendar sync works for me. Thanks a lot!!

**Comment by DrYak** [comment]
Calendar successfully tested as working for me too. 🎊

(I don't even use the contacts sync, I tend to simply reply to e-mail and fall back to LDAP to search for colleagues)

@jobisoft and @kxxt, you are now my heroes!

**Comment by fridrich** [comment]
OK, existing account from 91 with profile migrated to 102 and the two test extensions, works like a charm here and calendar syncs. Hotmail contacts are there even

**Comment by jobisoft** [comment]
Next needed steps:

- [ ] remove all overlaying code for the address book, that is a dead end
- [ ] no longer update individual card properties, but only the `_vCard` property (because we have more than two emails, which cannot be set through the legacy props)
- [ ] use code similar to what [the webExtension API](https://searchfox.org/comm-central/rev/1480d9f14070fee52641c2698fc9a104b02ceddb/mail/components/extensions/parent/ext-addressBook.js#1013-1120) is doing to update the _vCard prop
- [ ] alternatively, use the [WebExtension API](https://webextension-api.thunderbird.net/en/102/how-to/contacts.html) and use the [notifyTools](https://github.com/thundernest/addon-developer-support/tree/master/auxiliary-apis/NotifyTools) to access the background script and the WebExtension API directly from the legacy code still used by TbSync/EAS

**Comment by jretxaniz** [comment]
I only know how to add ans test the plugin from a .xpi file. How can I get it??

**Comment by blarghmatey** [comment]
I confirm that calendar syncing is now working again for me on 102.2.2 running on Manjaro Linux

**Comment by btaroli** [comment]
One presumes that we'll need to keep an eye on this issue, since XPI installed addons for testing won't auto-update later...

Still eagerly testing though. :D

**Comment by DavidBE-UK** [comment]
Works for me with TB102.2.2 with my normal profile (backed up first), finding and then syncing all my outlook.com calendars AND my outlook.com contacts. Great - thanks. Thought you said the contacts code isn't included - my outlook.com People has definitely appeared in the TB Address book and fully populated with the latest entries after syncing. (I had deleted the outlook.com account from TbSync back when it stopped working in TB 98-ish).

**Comment by jobisoft** [comment]
> Thought you said the contacts code isn't included - my outlook.com People has definitely appeared in the TB Address book and fully populated with the latest entries after syncing.

An initial download may work but I doubt later updates will work. The underlying contact storage has changed and for the initial download some migration code kicks in, but then the cards use the modern vCard storage and my addon currently only updates the old fields, which will be ignored by Thunderbird once a vCard has been created.

**Comment by jobisoft** [comment]
> I only know how to add ans test the plugin from a .xpi file. How can I get it??

XPI files are ZIP files, but I cannot dump XPI files here. You can install the ZIP file just like a XPI file, you do not even need to rename it.

**Comment by jamescr** [comment]
Calendar calendar sync is working again :tada:  
TbSync 3.1
EAS-4-TbSync 2.1
Thunderbird 102.1.2 (64-bit)
Debian 12 Bookworm (Testing) Linux

Thanks!!

**Comment by smmehta1822** [comment]
> Here is a very preliminary version for 102. Please do NOT use it on your production profile.
> 
> Create a new profile and test it - and assume that it will eat all your data, so have some sort of backup of your calendar and contacts data.
> 
> The address book UI has changed massively and I have not adapted it yet. This version aims for the following:
> 
>     * does calendar sync work on a new profile (up and down)
> 
>     * for the brave ones: does calendar sync work on an updated profile, using an existing sync state?
> 
> 
> [TbSync-beta.zip](https://github.com/jobisoft/EAS-4-TbSync/files/9581760/TbSync-beta.zip) [EAS-4-TbSync.zip](https://github.com/jobisoft/EAS-4-TbSync/files/9581763/EAS-4-TbSync.zip)

Thank you. I was able to configure it using TB 102.2 and Calendar sync with Exchange server is working again. Appreciate your help and support.

**Comment by jfvilasPlexus** [comment]
I've installed these zips  on a non-prod tb installation for testing purposes. initial sync went ok, but a second sync showed this error:

Unable to set DisplayName as a property, use vCardProperties

setProperty@resource:///modules/AddrBookCard.jsm:371:13
setProperty@chrome://tbsync/content/modules/addressbook.js:381:20
setThunderbirdItemFromWbxml@chrome://eas4tbsync/content/includes/contactsync.js:145:28
processCommands@chrome://eas4tbsync/content/includes/sync.js:933:53

**Comment by kxxt** [comment]
That error came from contract sync, which i
s not handled in this beta version.

**Comment by jobisoft** [comment]
Yep, you should all disable contact sync in this preview. It could be that TbSync blanks your contacts on the server.

**Comment by rol1** [comment]
Can you predict when we will see the update in addons.thunderbird.net?
Actuall I see there:
```
Find alternatives to no longer maintained Thunderbird extensions 
...
The add-on “TbSync” seems to be maintained, but a version for Thunderbird 102 has not yet been released. 
```

**Comment by jobisoft** [comment]
After @kxxt has taken care of the calendar sync, I am currently looking into the contact sync. I am not giving out ETAs anymore, but I want to get this done. I am currently working on the map between EAS properties and vCard properties.

@kxxt : If you are investigating this as well, please ping me, so we do not spend time on the same thing. I figured that staying in legacy code for now is the faster approach. I will manipulate the vCard directly, for everything that Thunderbird knows (suffix, title, phones, emails), and everything that is not shown in the UI I either use a dedicated X- value, or keep it as a value in the property bag.

**Comment by kxxt** [comment]
Personally I am not a user of contract sync so I might not be capable to develop that feature.

But I can do some code clean up because we no longer support old versions of ThunderBird.



> After @kxxt has taken care of the calendar sync, I am currently looking into the contact sync. I am not giving out ETAs anymore, but I want to get this done. I am currently working on the map between EAS properties and vCard properties.
> 
> @kxxt : If you are investigating this as well, please ping me, so we do not spend time on the same thing. I figured that staying in legacy code for now is the faster approach. I will manipulate the vCard directly, for everything that Thunderbird knows (suffix, title, phones, emails), and everything that is not shown in the UI I either use a dedicated X- value, or keep it as a value in the property bag.

**Comment by jim-davenport** [comment]
I am using the TBSync Beta posted here on 102.2.2. Everything seems to work with the exception of this sync error on my Hotmail contacts. I just thought it might help....

Unable to set DisplayName as a property, use vCardProperties

setProperty@resource:///modules/AddrBookCard.jsm:371:13
setProperty@chrome://tbsync/content/modules/addressbook.js:381:20
setThunderbirdItemFromWbxml@chrome://eas4tbsync/content/includes/contactsync.js:145:28
processCommands@chrome://eas4tbsync/content/includes/sync.js:933:53

**Comment by bjo81** [comment]
> I am using the TBSync Beta posted here on 102.2.2. Everything seems to work with the exception of this sync error on my Hotmail contacts. I just thought it might help....

It might help reading the previous comments as **contacts are not supported yet**.

**Comment by jim-davenport** [comment]
Apologies.

**Comment by cseader** [comment]
I have tested the beta released here with the zip files.

I am running: 
Thunderbird 102.2.2 
openSUSE Tumbleweed latest

Everything syncs perfectly, Thank You @jobisoft 

@jobisoft is there anything else you would like me to test for you?

**Comment by fdellwing** [comment]
Tested with Office365 and works fine (Contact sync disabled)

**Comment by ricounet67** [comment]
Tested with Office 365 and Exchange server. Both work

**Comment by jobisoft** [comment]
Here is the second preview, with [fixed contact sync](https://github.com/jobisoft/EAS-4-TbSync/commit/4556a8b9a99473d647f29f32cb2234f277f789f5).

[TbSync-beta.zip](https://github.com/jobisoft/EAS-4-TbSync/files/9610485/TbSync-beta.zip)
[EAS-4-TbSync.zip](https://github.com/jobisoft/EAS-4-TbSync/files/9610491/EAS-4-TbSync.zip)

You can install these zip files over the first preview (no need to uninstall). But you need both files.

**Make sure you have backups of your contacts and calendars (this preview also has updated calendar code).**

I recommend to first test a clean down sync of your contacts, so please disable all synced address books before trying this second preview. If you have a backup of your profile, you can test to sync an old sync state later.

The following things are worth testing:
* multiline fields: organisation, address street, notes 
* verifying up-sync of changes (TB to server web interface)
* verifying down-sync of changes (server web interface to TB)
* photos (EAS may reject too large photos)

There are a few things worth mentioning:
* The old AB fell short a lot of fields, which required to manipulate the AB UI.
* The new AB has a lot more fields, AB UI is no longer changed (but it is also no longer possible).
* The new organisation-position field has no counterpart in EAS (IIRC) - it is therefore not synced.
* The new AB UI allows more than 3 email addresses, but only three are supported in EAS - all other emails are not synced.
* The AB phone fields only support `Mobile`, `Home`, `Work`, `Fax` and `Pager` - EAS additionally supports `Assistant`, `Car`, `Radio`, `Company`. Those are displayed as `Unknown` with a prefix in the field itself. I have no idea how to do that better. Also note, that the EAS `HomeFax` field is mapped to TB `Fax`, while `BusinessFax` is mapped to `Unknown` with a prefix.
* The following EAS fields are no longer shown at all: `Spouse`, `ManagerName`, `AssistenName`. I have no idea where to put them.
* The 4 custom fields of TB are again mapped to `OfficeLocation`, `CustomerId`, `GovernmentId` and `AccountName` (those EAS fields seem not to been used very often and allowed to store the custom fields on the server).

If you have better ideas of how to manage all these special cases, tell me.

What is still not working in this preview:
* Autocomplete (? or does it work ?)
* global address book lookup

**Comment by jim-davenport** [comment]
For me, autocomplete is working with email addresses that have been used before, just not from the global address book.

**Comment by lguerard** [comment]
No errors when syncing contacts but no autocomplete and also nothing different in the address books ? I also only see the email addresse that have been used before.

**Comment by jobisoft** [comment]
> ... but no autocomplete ...

Are you referring to autocomplete from the global EAS address book (which was never synced, but only supported lookup) , or autocomplete from synced contacts?

> ... nothing different in the address books ...

What does that mean?

**Comment by lguerard** [comment]
Before TB102, when writing an email and starting typing a name, the plugin would show me email addresses from the Exchange server so I could select the one I wanted, now if the address wasn't already saved, nothing shows up. To answer your question, I'm guessing this is the global EAS address book. (Sorry not so familiar with all of this, I just knew that your plugin was helping a lot 😄 )

**Comment by jobisoft** [comment]
> Before TB102, when writing an email and starting typing a name, the plugin would show me email addresses from the Exchange server so I could select the one I wanted, now if the address wasn't already saved, nothing shows up. To answer your question, I'm guessing this is the global EAS address book. (Sorry not so familiar with all of this, I just knew that your plugin was helping a lot 😄 )

Yes, that is not yet working and will be addressed in the next preview. It will be announced here.

**Comment by sarashinai** [comment]
Many thanks to jobisoft for all hard work and emailing me directly to let me know about these threads.

I installed a clean download of 102 in a separate folder and installed the zip previews through the extension manager "Install from file". TbSync, EAS, and CalDav/CardDav all seemed to install fine but, when I tried to add an Exchange server through TbSync I got this:
![image](https://user-images.githubusercontent.com/73752895/191737081-7b23a719-4f2e-46cd-8e21-c5b7cf572266.png)

Did I install the wrong way or miss a step? Tried to read back through the posts but didn't spot anything that seemed to relate.

**Comment by jobisoft** [comment]
@sarashinai : Can you check in the add-on manager which version numbers you have installed of the 3 modules (tbsync, eas provider & dav provider)?

**Comment by mdiskuze** [comment]
> Here is the second preview, with [fixed contact sync](https://github.com/jobisoft/EAS-4-TbSync/commit/4556a8b9a99473d647f29f32cb2234f277f789f5).
> 
> [TbSync-beta.zip](https://github.com/jobisoft/EAS-4-TbSync/files/9610485/TbSync-beta.zip) [EAS-4-TbSync.zip](https://github.com/jobisoft/EAS-4-TbSync/files/9610491/EAS-4-TbSync.zip)
> 
> You can install these zip files over the first preview (no need to uninstall). But you need both files.
> ...

Hi, many thanks for your great work! But I have problem with contact sync - "Failed to convert WBXML to XML!". No contact were downloaded. I've tried completely remove all contacts and cleanup mi address book, but problem is same. Any idea why? Thanks.

log without real names and keys:
```
** Fri Sep 23 2022 10:19:22 GMT+0200 **
[setSyncState] : State: preparing, Account: Foo Bar, Folder: Contacts

** Fri Sep 23 2022 10:19:22 GMT+0200 **
[setSyncState] : State: prepare.request.estimate, Account: Foo Bar, Folder: Contacts

** Fri Sep 23 2022 10:19:22 GMT+0200 **
[setSyncState] : State: send.request.estimate, Account: Foo Bar, Folder: Contacts

** Fri Sep 23 2022 10:19:22 GMT+0200 **
[WBXML: Sending data <send.request.estimate> for Foo Bar (Contacts)] : 
3 1 6a 0 0 6 45 47 48 0 0 4b 3 31 30 37 30 31 39 34 32 38 30 0 1 0 6 4a 3 31 36 0 1 0 0 57 50 3 43 6f 6e 74 61 63 74 73 0 1 1 0 6 1 1 1

** Fri Sep 23 2022 10:19:22 GMT+0200 **
[XML: Sending data <send.request.estimate> for Foo Bar (Contacts)] : 
<?xml version="1.0"?>
<GetItemEstimate xmlns='GetItemEstimate'>
<Collections>
<Collection>
<SyncKey xmlns='AirSync'>1070194280</SyncKey>
<CollectionId>16</CollectionId>
<Options xmlns='AirSync'>
<Class xmlns='AirSync'>Contacts</Class>
</Options>
</Collection>
</Collections>
</GetItemEstimate>

** Fri Sep 23 2022 10:19:22 GMT+0200 **
[Sending (EAS v14.0)] : POST https://outlook.office365.com/Microsoft-Server-ActiveSync?Cmd=GetItemEstimate&User=foo.bar%40foobar.com&DeviceType=TbSync&DeviceId=...

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[WBXML: Receiving data <send.request.estimate> for Foo Bar (Contacts)] : 
3 1 6a 0 0 6 45 4d 4e 3 31 0 1 48 4a 3 31 36 0 1 4c 3 30 0 1 1 1 1

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[XML: Receiving data <send.request.estimate> for Foo Bar (Contacts)] : 
<?xml version="1.0"?>
<GetItemEstimate xmlns='GetItemEstimate'>
<Response>
<Status>1</Status>
<Collection>
<CollectionId>16</CollectionId>
<Estimate>0</Estimate>
</Collection>
</Response>
</GetItemEstimate>

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[setSyncState] : State: eval.response.estimate, Account: Foo Bar, Folder: Contacts

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[Extracted XML data] : 
{"GetItemEstimate":{"Response":{"Status":"1","Collection":{"CollectionId":"16","Estimate":"0"}}}}

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[setSyncState] : State: prepare.request.remotechanges, Account: Foo Bar, Folder: Contacts

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[setSyncState] : State: send.request.remotechanges, Account: Foo Bar, Folder: Contacts

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[WBXML: Sending data <send.request.remotechanges> for Foo Bar (Contacts)] : 
3 1 6a 0 45 5c 4f 4b 3 31 30 37 30 31 39 34 32 38 30 0 1 52 3 31 36 0 1 1e 13 55 3 35 30 0 1 57 50 3 43 6f 6e 74 61 63 74 73 0 1 0 11 45 46 3 31 0 1 1 0 0 1 1 1 1

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[XML: Sending data <send.request.remotechanges> for Foo Bar (Contacts)] : 
<?xml version="1.0"?>
<Sync xmlns='AirSync'>
<Collections>
<Collection>
<SyncKey>1234567890</SyncKey>
<CollectionId>16</CollectionId>
<DeletesAsMoves/>
<GetChanges/>
<WindowSize>50</WindowSize>
<Options>
<Class>Contacts</Class>
<BodyPreference xmlns='AirSyncBase'>
<Type xmlns='AirSyncBase'>1</Type>
</BodyPreference>
</Options>
</Collection>
</Collections>
</Sync>

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[Sending (EAS v14.0)] : POST https://outlook.office365.com/Microsoft-Server-ActiveSync?Cmd=Sync&User=foo.bar%40foobar.com&DeviceType=TbSync&DeviceId=...

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[WBXML: Receiving data <send.request.remotechanges> for Foo Bar (Contacts)] : 


** Fri Sep 23 2022 10:19:23 GMT+0200 **
[XML: Receiving data <send.request.remotechanges> for Foo Bar (Contacts)] : 
Failed to convert WBXML to XML!


** Fri Sep 23 2022 10:19:23 GMT+0200 **
[setSyncState] : State: prepare.request.localchanges, Account: Foo Bar, Folder: Contacts

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[setSyncState] : State: done, Account: Foo Bar```

**Comment by jobisoft** [comment]
```
** Fri Sep 23 2022 10:19:23 GMT+0200 **
[WBXML: Receiving data <send.request.remotechanges> for Foo Bar (Contacts)] : 

** Fri Sep 23 2022 10:19:23 GMT+0200 **
[XML: Receiving data <send.request.remotechanges> for Foo Bar (Contacts)] : 
Failed to convert WBXML to XML!
```

The provider does not expect to receive an empty response. 

Can you identify which address book this is and disable it (not sync it)?
Do the other address books work once you disabled the broken one?
Is the broken address book empty?
This is caused by unchanged code, I do not think this could have worked in TB91, did it?

**Comment by sarashinai** [comment]
> @sarashinai : Can you check in the add-on manager which version numbers you have installed of the 3 modules (tbsync, eas provider & dav provider)?

CalDav: 2.2
EAS: 2.1
TbSync: 3.2

**Comment by jobisoft** [comment]
EAS 2.1 is preview 1, while TbSync 3.2 is preview 2 (or you got it from the DAV repo).

You need EAS from preview 2:
https://github.com/jobisoft/EAS-4-TbSync/issues/167#issuecomment-1252806380

**Comment by sarashinai** [comment]
> EAS 2.1 is preview 1, while TbSync 3.2 is preview 2 (or you got it from the DAV repo).
> 
> You need EAS from preview 2: [#167 (comment)](https://github.com/jobisoft/EAS-4-TbSync/issues/167#issuecomment-1252806380)

Now have EAS 2.2 but issue remains.

**Comment by jobisoft** [comment]
Huh, there was a config error indeed.

I have released the final versions for Tbsync and DAV in the 4.x branch:
https://github.com/jobisoft/TbSync/releases/download/v4.1/TbSync.xpi
https://github.com/jobisoft/DAV-4-TbSync/releases/download/v4.1/DAV-4-TbSync.xpi

This is a pre-release of EAS:
https://github.com/jobisoft/EAS-4-TbSync/releases/download/v4.1/EAS-4-TbSync.xpi

These 3 versions should play together now.

**Comment by sarashinai** [comment]
> 

Confirmed those work together. Now testing functionality. Thank you!

**Comment by mdiskuze** [comment]
> ```
> ** Fri Sep 23 2022 10:19:23 GMT+0200 **
> [WBXML: Receiving data <send.request.remotechanges> for Foo Bar (Contacts)] : 
> 
> ** Fri Sep 23 2022 10:19:23 GMT+0200 **
> [XML: Receiving data <send.request.remotechanges> for Foo Bar (Contacts)] : 
> Failed to convert WBXML to XML!
> ```
> 
> The provider does not expect to receive an empty response.
> 
> Can you identify which address book this is and disable it (not sync it)? Do the other address books work once you disabled the broken one? Is the broken address book empty? This is caused by unchanged code, I do not think this could have worked in TB91, did it?

My mistake, I accidentally synced an empty calendar. The error message didn't look like it, but once I synchronized the correct address book - everything went fine. I apologize for my mistake and thanks for your help.

**Comment by jobisoft** [comment]
Well, the provider is still showing a somewhat buggy behaviour in this case. If the calendar is empty, it should simply create an empty calendar and not crash like that. Did it create an empty calendar?

Could you describe a bit more how you got to this error? Is it like so?
- you have an empty calendar in our EAS account
- you asked TbSync to sync that calendar to Thunderbird
- Thunderbird aborted the sync and showed an error (which error?)
- You went to the debug log to get the details?

**Comment by jim-davenport** [comment]
All is well with the exception of auto-completing email addresses from our global directory. Is that because EAS is still in preview? I'm not exactly sure what TbSync does vs. EAS. Sorry. And THANKS for your hard work!

**Comment by DavidBE-UK** [comment]
Problem. I noticed in the EAS-4-TbSync version of my Outlook.com contacts that for the one contact in it where there is a work email and a home email address both emails are there but, unlike with the work and home phone numbers, it didn't show the work and home labels, just the two email addresses, So, in the TB AB I selected the work and home types, then tried to sync. This produced errors, from the event log:

Bad item skipped: JavaScriptError

Account: Outlook (EAS)
Resource: Contacts
(Name of contact and the two email addresses)

and a second error entry:

JavaScriptError!

(Account and resource as above)

raw.join is not a function

getWbxmlFromThunderbirdItem@chrome://eas4tbsync/content/includes/contactsync.js:449:37
getWbxmlFromThunderbirdItem@chrome://eas4tbsync/content/includes/sync.js:1121:55
sendLocalChanges@chrome://eas4tbsync/content/includes/sync.js:564:85

It kept repeating this, and continued after taking the contact back to how it was (removing the home and work labels).

In the end I had to disable the account, then re-enable so it re-synced the contacts from scratch.

I now notice that whereas the Windows 10 People does include the ability to label email addresses as home and work (which I had), the web interface to the oulook.com People does not, so presumably the problem is EAS-4-TbSync trying to send parameters with the email addresses that outlook.com doesn't support. However, it should handle this in a way that doesn't create an error that needs the whole AB disabling then re-enabling and re-importing.

(Also, it would be nice if the event log allowed the user to select and copy the whole entry, not just the JavaScript part, and allowed access to copy in a context menu for the JavaScript part for people like me who don't normally use keyboard 'shortcuts' to copy.)

**Comment by jobisoft** [comment]
> Is that because EAS is still in preview? I'm not exactly sure what TbSync does vs. EAS. Sorry.

I have not yet re-implemented global address book lookup (GAL) for EAS. This will need a few more weeks. I will probably release the provider without this functionality and not delay this until GAL is done.

**Comment by jim-davenport** [comment]
> I have not yet re-implemented global address book lookup (GAL) for EAS. This will need a few more weeks. I will probably release the provider without this functionality and not delay this until GAL is done.

Understood. Thank you again! Off-topic - I was wondering if the functionality provided by TbSync and EAS will be included in future versions of Thunderbird. I've always been confused as to why extensions are needed.

**Comment by jobisoft** [comment]
> Off-topic - I was wondering if the functionality provided by TbSync and EAS will be included in future versions of Thunderbird. I've always been confused as to why extensions are needed.

EAS is not a standard protocol, and Mozilla has a history of supporting and pushing standards. By supporting EAS (or any of the other Microsoft can-of-worms), they would help to establish this "everybody is doing their own soup" instead of working together creating a well working standard.

**Comment by paulastronaut** [comment]
I have the same error message as [https://github.com/jobisoft/EAS-4-TbSync/issues/167#issuecomment-1256217675](url) when syncing contacts after a change in Thunderbird:

raw.join is not a function

getWbxmlFromThunderbirdItem@chrome://eas4tbsync/content/includes/contactsync.js:449:37
getWbxmlFromThunderbirdItem@chrome://eas4tbsync/content/includes/sync.js:1121:55
sendLocalChanges@chrome://eas4tbsync/content/includes/sync.js:564:85

Syncing contacts from scratch is working o.k., but if I change contacts in Thunderbird, I often (not always) get this error. I did some tests with different contacts and different fields, sometimes syncing works, sometimes not. I suspect that it is dependent on the specific contact I change whether syncing works or throws that error.

**Comment by jobisoft** [comment]
@DavidBE-UK @paulastronaut : Please try:
[EAS-4-TbSync.zip](https://github.com/jobisoft/EAS-4-TbSync/files/9637158/EAS-4-TbSync.zip)

This is not fixing it, but adding a better error description. What do you get?

**Comment by paulastronaut** [comment]
Thanks for looking into this. New error message:

Failed to eval value: <"Edinburghplatz 7"> @ {"item":"adr","type":"text","params":{"type":"work"},"index":2}

getWbxmlFromThunderbirdItem@chrome://eas4tbsync/content/includes/contactsync.js:453:31
getWbxmlFromThunderbirdItem@chrome://eas4tbsync/content/includes/sync.js:1121:55
sendLocalChanges@chrome://eas4tbsync/content/includes/sync.js:564:85

**Comment by DavidBE-UK** [comment]
1st entry same, second is now:

Failed to eval value: <"103 Stoke Road"> @ {"item":"adr","type":"text","params":{"type":"home"},"index":2}

getWbxmlFromThunderbirdItem@chrome://eas4tbsync/content/includes/contactsync.js:453:31
getWbxmlFromThunderbirdItem@chrome://eas4tbsync/content/includes/sync.js:1121:55
sendLocalChanges@chrome://eas4tbsync/content/includes/sync.js:564:85

103 Stoke Road is the first line of the postal address ... odd as that wasn't a problem before. The only change I made to the contact in the TB AB before trying to sync and getting this error was adding the 'Home' and 'Work' tags to the home and work email addresses, no change to the postal address. As I said before, it now appears that, unlike the GMail contacts, the outlook.com ones don't support the tags on the email addresses. If outlook.com isn't expecting them, maybe that's causing things to be sent in what outlook.com thinks is the wrong order?

**Comment by jobisoft** [comment]
Does the address street field have a second line? I expect an array with all the lines street in the street field actually. Strange.

**Comment by DavidBE-UK** [comment]
No. 1 address line, then city field. (The edit box is sized as if their could be a second line.)

**Comment by paulastronaut** [comment]
More tests: As soon as I add a street in an contact (irrevelant if private, business, or unknown address), the error occurs. Adding other fields (postal code, city) without a street is fine. The street in my case is just one line. It doesn't matter if street consists of several parts (e. g. street name and house number, separated by blank) or if it is just one word.

**Comment by DavidBE-UK** [comment]
I just added an extra line to the street address (just a line with a comma, nothing else) and it synced without error. It appears EAS-4-TbSync is expecting two lines in the address street field, and is erroring if there's only one.

**Comment by jobisoft** [comment]
what happens if the street field has multiple lines?

**Comment by paulastronaut** [comment]
I can confirm that adding a second line to the street field makes the error go away.

**Comment by jobisoft** [comment]
Can you confirm this is fixed in EAS 4.1.2?
[EAS-4-TbSync.zip](https://github.com/jobisoft/EAS-4-TbSync/files/9638174/EAS-4-TbSync.zip)

**Comment by DavidBE-UK** [comment]
Yes, I can confirm it is now syncing one line addresses without an error. Thanks.

**Comment by jobisoft** [comment]
Are we save to release 4.1.2 as a public version and start submitting it to addons.thunderbird.net? Or should I declare it as pre-release again?

**Comment by DavidBE-UK** [comment]
If you'd asked and I'd answered immediately after I tried that latest version this morning, I'd have said it was fine as far as I'm concerned. Unfortunately I'm now not at all confident of anything re. the TB Address Book and any of the separate address books in it, let alone just TbSync/EAS-4-TbSync. I don't actually use the TB AB much, and I've only just noticed my general Address Book situation between laptop programs, cloud and phone apps is weirdly messed up in at least two different ways and I'm struggling to understand it. One problem is just in TB but is actually affecting not only the outlook.com synced TB AB but also the gmail.com synced one AND the Personal AB all the same, so that must be a basic TB problem, not an TbSync/EAS-4-TbSync problem. The other problem does just affect the outlook.com address book, but is weird because the TbSync/EAS-4-TbSync outlook.com address book in TB and the Windows 10 People address book, both synced to my outlook.com account, agree with each other and show what they should, but the actual outlook.com address book displayed in my browser is messed up and is NOT displaying the right thing for 3 entries and I can't figure out why or how to fix it. But I _think_ it's a problem with the outlook.com address book web interface rather than with TbSync/EAS-4-TbSync. So I think TbSync/EAS-4-TbSync is OK for release, but I'm not 100% sure about anything address book related at the moment. Sorry.

**Comment by paulastronaut** [comment]
Version 4.1.2 solves the address issue for me as well, synchronizing street names containing one line works now. As to if this should be a public version, I don't feel confident to judge this. I did no intensive testing. Things look good for me at the moment, but I had an issue until yesterday with an error message "unexpected empty server response" (or similar, I don't remember the exact wording). The error doesn't show up anymore at the moment, perhaps due to an update of the backend (SOGo).

**Comment by jobisoft** [comment]
OK, I will release it as a pre-release, replacing 4.1.

What other issues are you observing (with all your address books)? For each address book please apecify how it is synced (or local only).

The Sogo AB is also an EAS AB? Or a CardDAV AB? Best is to make a screenshot of the error, of it happens again.

Thanks for your testing!

**Comment by DavidBE-UK** [comment]
Well, for historical reasons, including coming late (2017) to both Thunderbird (the email client I used before had an address book that didn't sync to anything external) and having a smart phone (also 23017), being confused about what was best to sync to what between Windows 10 laptop and android Phone, and never quite settling deciding whether I should principally base address books and calenders on my Microsoft account or my GMail one, I have a ... clumsy situation.

I only keep 56 contacts (my social and work lives ended when I became disabled in 2007), some only phone number, some only email address, some only postal address, some two out of the three, a few all three. In TB I have 3 address books (not counting Collected Addresses): Personal; Outlook, synced to my outlook.com People; and Google, synced to my gmail.com Contacts. My laptop Windows 10 People also syncs to both outlook.com PEople and gmail.com Contacts, And on my Android phone, I have both Google Contacts and the Outlook app both syncing with both the outlook.om People and the gmail.com Contacts, plus a 3rd party Contacts App and the FairEmail app. (i have the same tangle with calendars.) Brain fog side effects from permanent presciption painkillers (3 different all the time) means i can't think, follow or diagnose things clearly, and yes, this is all a tangle where when I go to add contacts or calendar entries I'm never quite sure where things will turn up after syncing. In theory, I principally add new contacts and calender entries to the Windows 10 People and Calendar apps on my laptop, and hope they then sync to Thunderbird and my phone apps.

In theory my Personal, Outlook and Google address books in Thunderbird should all currently contain the same 56 contacts, give or take the typical Google gmail.com address book field differences. I normally add new contacts into the Windows 10 People, and when that's synced to Thunderbird I'll copy the new contacts from the outlook to the Personal and Google address books.

I now have noticed that in all three address books - Personal, Outlook, Google - in the horizontal view of the list of contacts I use, in the (postal) address book column, in  the ones with postal addresses some - but not all - of the addresses that have two lines (before the city field) are showing them with the second line BEFORE the first line, even though they are in the right order in the lower pane showing the whole contact. But as this is affecting the Google address book and especially the personal address book as well as the Outlook one, I can't see it can be anything to do with TbSync (unless it's doing something to all the address books).

My Outlook.com specific problem is that I have 7 email addresses, for which I have 7 separate contacts, but while they are showing correctly in the TbSync/EAS-4-TbSync address book in TB, AND in the outlook.com address book in the Windows 10 People. even after syncing both, 3 of the contacts in outlook.com in a browser are not showing the different email addresses they are supposed to, but are all showing the same one of the other email addresses instead. And if I correct them outlook.com in the browser then sync, I end up with a second email address (same as the correct one already displaying) in the TB Outlook address book. And if I then delete the extra one in TB and sync, the corrected one disappears and the previous wrong one reappears in the browser outlook.com. I can't fathom how outlook.com came to show the wrong  email addresses, or how the synced TB and W10 People can be showing the different correct values.

**Comment by DavidBE-UK** [comment]
Oh. Just had to go and do something, came back, looked at my TB AB and things have changed. The 3 address books - Outlook, Google and Personal - are now, after syncing while I was away, presumably, all different in which have or haven't got reversed postal address lines. And in the Outlook one synced by TbSync several contacts have lost the second lines of postal addresses completely AND they've also gone from the outlook.com People in the browser. And they've gone from the W10 People app too.

**Comment by wonder75** [comment]
Hallo, will the manually installed preview versions update to the release version with the thunderbird updater?

Was wondering if I should migrate now or better wait to not have to do things twice.

Thanks for your work! Really appreciated.

**Comment by jobisoft** [comment]
> Hallo, will the manually installed preview versions update to the release version with the thunderbird updater?

Yes. When there is a new version on addons.thunderbird.net, you will get it regardless which version you have currently installed.

**Comment by jobisoft** [comment]
> Oh. Just had to go and do something, came back, looked at my TB AB and things have changed. The 3 address books - Outlook, Google and Personal - are now, after syncing while I was away, presumably, all different in which have or haven't got reversed postal address lines. And in the Outlook one synced by TbSync several contacts have lost the second lines of postal addresses completely AND they've also gone from the outlook.com People in the browser. And they've gone from the W10 People app too.

That sounds very odd. What should have changed your personal AB? It is local, is it not? How is your GMail AB connected, via CardDAV? How are GMail and Outlook.com connected? How should GMail pick up changes from outlook?

The TbSync debug log (if enabled) has the detailed log of incoming changes. So you could at least see why the outlook.com AB has changed. If there is a change request coming from the server (for whatever reason, than TbSync has to follow that.

**Comment by paulastronaut** [comment]
> The Sogo AB is also an EAS AB? Or a CardDAV AB?

Sogo provides an ActiveSync backend: [https://www.sogo.nu/files/docs/SOGoInstallationGuide.html#_microsoft_enterprise_activesync](url) It works quite well, IMHO.

**Comment by jobisoft** [comment]
Caution: I think for some reason I am eating email addresses.

**Comment by jobisoft** [comment]
Ouch. That will take a while. PLease disable your EAS address books. I fail to pick the correct email and very often simply send none at all to the server. Most servers interpret that as "no change in emails", others take that as a clear instruction.

**Comment by jobisoft** [comment]
I have fixed the email issue. It happened if you manually switched an email from "undefined" to "home" or "work" in Thunderbird, and then synced that back to the EAS server. The provider then failed to "see" those emails and did not send them to the server.

For everybody testing the pre releases:
- disable all your address books in the TbSync manager 
- install v4.1.3 of the EAS provider (https://github.com/jobisoft/EAS-4-TbSync/releases/download/v4.1.3/EAS-4-TbSync.xpi)
- enable your address books again, to start a clean sync

**Comment by paulastronaut** [comment]
Thanks for the update. Did some tests with v4.1.3 by adding e-mail-addresses and changing their type. Everything (not only address, but all other syncs) looks good so far, the add-in is doing its job smoothly AFAICT.

**Comment by rqi14** [comment]
Hi. My calendar does not sync. The events I created locally could not be uploaded to outlook. The error is:
** Mon Sep 26 2022 01:44:55 GMT+0100 (�����) **
[XML: Receiving data <send.request.remotechanges> for ******@live.com (Calendar)] : 
Failed to convert WBXML to XML!

**Comment by jobisoft** [comment]
> Hi. My calendar does not sync. The events I created locally could not be uploaded to outlook. The error is: ** Mon Sep 26 2022 01:44:55 GMT+0100 (������) ** [XML: Receiving data <send.request.remotechanges> for ******@live.com (Calendar)] : Failed to convert WBXML to XML!

1. Did it work with TB102?
2. Arre you using EAS v4.1.3?
3. What exact error do you get in the TbSync manager?
4. Is just one calendar falling, or all?

**Comment by wonder75** [comment]
Tested the latest preview version with Horde Groupware.
Calendar and Contacts seem to sync just fine.
Don't use Tasks.

**Comment by rarunkarthik** [comment]
I am using TB 102 on ubuntu. I have office365 account and the contact list from my account's Contact list is showing up, but the contacts from the GAL is not showing up.

**Comment by jobisoft** [comment]
GAL is not implemented for 102. GAL has been removed and will come back later.

**Comment by jim-davenport** [comment]
If you had to guess… when will GAL be back?

**Comment by jobisoft** [comment]
So there is another show stopper: I am flipping Org:Company Name and Org:Department if both are present. This is due to some very strange Thunderbird implementation: If only one line is given in the Org field, it is returned as a string (presumably the Company Name), if both are present, it returns an array with Department at index 0 and Company Name and index 1.

\o/

Disable contact sync until this is fixed.

**Comment by jobisoft** [comment]
v4.1.4 is available. As usual:

* disable contact sync if not already done
* install https://github.com/jobisoft/EAS-4-TbSync/releases/tag/v4.1.4
* re-enable contact sync

Thanks for testing. Hope we are ready to release soon.

**Comment by lguerard** [comment]
I just installed 4.1.4 but TbSync tells me that `Provider "Exchange ActiveSync" for TbSync is not yet installed`, while it was fine with the previous version...

**Comment by jobisoft** [comment]
@lguerard : Can you check if you are also running the latest release of TbSync?
https://github.com/jobisoft/TbSync/releases/

**Comment by jobisoft** [comment]
Can you please both check if you are running the most recent version of TbSync?
https://github.com/jobisoft/EAS-4-TbSync/issues/167#issuecomment-1260692244

EAS 2.0.1 definitely is using API schema 2.4 while EAS 4.* is using API schema 2.5. This is a strong indicator, that you are not using the latest release of TbSync, which also uses API schema 2.5 (and would not work with EAS 2.0.1)

**Comment by rubennc91** [comment]
Thank you very much.

I thought I had the latest version, but I was using some intermediate version. I'm sorry. 

I am using Windows 10 64-bit, TB 102.3.0, TbSync 4.1, EAS-4-TbSync 4.1.4.

**Comment by lguerard** [comment]
All good now, thanks for the pointer and sorry about this ! :)

**Comment by paulastronaut** [comment]
If I create a calendar event in Thunderbird with an invitation to A, then sync it with EAS, and then look it up on my phone, the invitation to A is missing. If I add B to the same meeting on my phone and sync the calendars, the invitation to B gets added to Thunderbird, but then in Thunderbird the invitation to A is missing. Can somebody please check if this is occuring only in my setup?

**Comment by jobisoft** [comment]
My EAS provider has no good invite/meeting support. It could work, but I know of many things which can go wrong. You should use your servers webinterface to create/accept invites.

**Comment by rarunkarthik** [comment]
I have the latest EAS-4-TbSync as well as TbSync addons as of today (Oct 1st) but still no luck with Office 365 GAL contacts.

**Comment by jobisoft** [comment]
https://github.com/jobisoft/EAS-4-TbSync/issues/167#issuecomment-1258488307

**Comment by johnjohnk2** [comment]
Tested and seems to work fine here . Many thanks @jobisoft and @kxxt

**Comment by groentebroer** [comment]
I do not know if this is the right place, but I am using TBSync 4.1 and Provider for EAS 4.1.4.

Sync on the contacts fails with the following error. (I only sync my personal Contacts).
Expanding the debug log does not seem to provide more info.

If I remember correctly the errors came when I added a contact locally (or update one).

Log excerpt (anonimized where needed):
```** Mon Oct 03 2022 20:34:07 GMT+0200 (Midden-Europese zomertijd) **
[WBXML: Receiving data <send.request.remotechanges> for AAA O365 (Contacts)] : 


** Mon Oct 03 2022 20:34:07 GMT+0200 (Midden-Europese zomertijd) **
[XML: Receiving data <send.request.remotechanges> for AAA O365 (Contacts)] : 
Failed to convert WBXML to XML!


** Mon Oct 03 2022 20:34:07 GMT+0200 (Midden-Europese zomertijd) **
[setSyncState] : State: prepare.request.localchanges, Account: AAA O365, Folder: Contacts

** Mon Oct 03 2022 20:34:07 GMT+0200 (Midden-Europese zomertijd) **
[EventLog] : BadItemSkipped::JavaScriptError
Surname, Name (Name, Surname) <name.name@domain.nl>

** Mon Oct 03 2022 20:34:07 GMT+0200 (Midden-Europese zomertijd) **
[EventLog] : JavaScriptError
undefined

undefined
```

**Comment by meinhardh** [comment]
On my Ubuntu 20.04 system, an apt dist-upgrade this morning boosted TB from 91.11.0 to 102.2.2. Of course only afterwards I realised that tbsync and the EAS provider are not compatible. I have since read through the all important threads, and understand that
- Running pre-releases of tbsync and the EAS provider is strongly discouraged (risk of severe data loss)
- Downgrading TB to 91 is strongly discouraged (risk of severe data loss)
- TB 102 is currently not able to sync with Exchange
- Resolving the situation depends on activities outside the tbsync/EAS provider realm
If this is all true, it's a pretty bad deadlock/catch-22 situation forcing me to use braindead OWA for calendaring and task lists...
Can somebody confirm or tell me wrong, please?

Having said that, there is a flip side to the hard hit now - tbsync and Co. have worked marvellously well so far - big kudos and thanks to all involved for the hard, tremendously successful work!

**Comment by jobisoft** [comment]
The reason for not releasing TbSync (the main manager add-on) is the categories bug, where Thunderbird eats categories stored in vCards. TbSync/EAS currently stores categories in a custom property and is not affected. TbSync/DAV is affected 8and every manually setup CardDAV address book in Thunderbird). If you do not use TbSync/DAV, you are not affected.

TbSync/EAS is not released, because I still want to have more feedback if everything works. Backup your data and give it a go. 

However, I will not release TbSync/EAS before TbSync is released, and ship them all in one go.

**Comment by meinhardh** [comment]
Thanks, that clarifies things. I've installed tbsync v4.1 and EAS-4-TbSync v4.1.4. So far things look good indeed. (On top of Exchange calendars, I'm using a single DAV calendar directly from TB, no vCards involved.)

**Comment by tigerjack** [comment]
I've installed tbsync v4.1 and EAS-4-TbSync v4.1.4 as well. Things looks great up to now.

**Comment by edlerluca** [comment]
> TbSync/EAS is not released, because I still want to have more feedback if everything works. Backup your data and give it a go.

I'm using tbsync 4.1 + EAS 4.1.4 with multiple Clients (Windows 10 and Ubuntu 20.04, both with newest Thunderbird 102.3.1) together with Outlook/Exchange 365 since about two weeks and everything works just perfectly. There hasn't been a single issue. Thanks for your great work!

Just before this comment I upgraded the EAS from 4.1.x to 4.1.4. Sync still works.

**Comment by jobisoft** [comment]
See https://github.com/jobisoft/EAS-4-TbSync/issues/167#issuecomment-1268126685

I will announce the official release here.

**Comment by smmehta1822** [comment]
Thank you.

**Comment by rmknust** [comment]
Sounds like some guys spend hours & weeks on solving this...Chapeau, wonderful!!!

The only -maybe silly- question as a simple Linux Mint 20.3 Una & Thunderbird User: from where and how do I get the updates TbSync4.1 EAS 4.1.4 installed? Nothing found on Thunderbird Add-On Manager...

**Comment by edlerluca** [comment]
> The only -maybe silly- question as a simple Linux Mint 20.3 Una & Thunderbird User: from where and how do I get the updates TbSync4.1 EAS 4.1.4 installed? Nothing found on Thunderbird Add-On Manager...

The prerelease downloads for Tbsync, EAS and DAV are available here https://github.com/jobisoft/EAS-4-TbSync/issues/167

If you installed a previous version manually, just go to the Thunderbird addon menu, select the plugin, in the upper right the “gear”, select “install from file” and select the updated file you just downloaded. It overrides the existing addon installation and keeps all settings.

**Comment by rmknust** [comment]
WOW...thanks for super fast help...looks like it works...all installed, 
all initially synced, no error messages :-) Later I will check if 
adds/changes/deletes are synced well, too

Once again, thanks for help and crystal clear instruction :-)))

Michael

**Comment by JonasKonrad** [comment]
Hi folks,
hi @jobisoft 

just upgraded to EAS 4.1.4, TbSync 4.1, Thunderbird 102.2.2 64b on Ubuntu 20.04.5. Had a working O365 Exchange Calendar before, being deactivated by a TB update (that is, via the resulting AddOn deactivation). For curiosity and not directly seeing the release Assets, I used the Makebeta file to create the xpi files. Everything turned out to be working:

-  "Old" items remain stably accesible
-  New items get synced (created on another device before the first sync with TB EAS).

Cheers,
Jonas

**Comment by wol-win** [comment]
> I have released the final versions of TbSync and of the DAV provider in their 4.x branches: https://github.com/jobisoft/TbSync/releases/ https://github.com/jobisoft/DAV-4-TbSync/releases/
> 
> This is a pre-release of the EAS provider: https://github.com/jobisoft/EAS-4-TbSync/releases/
> 
> More information about the state of the EAS pre-release can be found here: [#167 (comment)](https://github.com/jobisoft/EAS-4-TbSync/issues/167#issuecomment-1252806380)
> 
> **TbSync/EAS is NOT ready for productive use!**
> 
> **NOTE: To avoid data loss, I will release TbSync for Thunderbird 102 only after https://bugzilla.mozilla.org/show_bug.cgi?id=1792399 has been backported to Thunderbird 102.**

Is this backporting to TB 102 expected to happen, or do we need to wait until TB106 is released?

**Comment by jobisoft** [comment]
I expect a backport soon.

**Comment by ghost-from-the-past** [comment]
just to confirm it works ok (great!!!) for me
I downloaded the last commit from github (boyh TBSyn and EAS) and installed
on
MacOS 12.6 + TB 102.3.1
Ubuntu 22.04.1 + TB 102.3.1

thanks for your fantastic work!!!

**Comment by myxpt2** [comment]
Thanks for all your work @jobisoft. Working well here with MacOS / TB 102.3.0.

**Comment by jim-davenport** [comment]
If you had to estimate... When do you think GAL will be ready. Everything else is working great!

**Comment by MengXiangxi** [comment]
Hi all, I am just here to provide feed back that I am having no issue with my TB 102.3.1 on Windows 10. I am using TbSync 4.1 + TbSync EAS 4.1.4. I am mainly syncing my Exchange account.

**Comment by felixoi** [comment]
Working as expected for me on TB 102.3.1

**Comment by floschl** [comment]
Win10+TB102.3.**2**+TbSync4.1+EAS 4.1.4+Outlook365 
works very smoothly (bi-directional calender+contact syncs)

thank you very much for your efforts!!

**Comment by kla-s** [comment]
Hi all, 
first thanks @jobisoft for the great work and support!

Actually i am currently struggling to get the pre releases to work. I'm on Ubuntu 22.04.1 LTS, Kernel 5.15.0-48-generic. Installed TB102.3.3-1 via snap latest/candidate, [TbSync v4.2 ](https://github.com/jobisoft/TbSync/releases/download/v4.2/TbSync.xpi) and [EAS-4-TbSync v4.1.4](https://github.com/jobisoft/EAS-4-TbSync/releases/download/v4.1.4/EAS-4-TbSync.xpi).
When trying to install TbSync v4.2 TB says its not compatible with TB102.3.3, so im stuck there. The [release page](https://github.com/jobisoft/TbSync/releases/) says release v4.2 is for Thunderbird. 

Just for completeness: EAS-4-TbSync v4.1.4 does install but is obv not helpfull w/o TbSync.

Any pointers would be great, other than that i wanted to share my differing experience for completeness sake

**Comment by smmehta1822** [comment]
@jobisoft  Thank you for the clarity and work and support you have provided.

**Comment by ghost-from-the-past** [comment]
at @kla-s 
> Actually i am currently struggling to get the pre releases to work. I'm on Ubuntu 22.04.1 LTS, Kernel 5.15.0-48-generic. Installed TB102.3.3-1 via snap latest/candidate, [TbSync v4.2 ](https://github.com/jobisoft/TbSync/releases/download/v4.2/TbSync.xpi) and [EAS-4-TbSync v4.1.4](https://github.com/jobisoft/EAS-4-TbSync/releases/download/v4.1.4/EAS-4-TbSync.xpi). When trying to install TbSync v4.2 TB says its not compatible with TB102.3.3, so im stuck there. The [release page](https://github.com/jobisoft/TbSync/releases/) says release v4.2 is for Thunderbird 102.3.3.

you can just create the xpi file by downloading the last commit
the xpi is just a zip file
(you can check one of the xpi files in your TB profile /extensions, just rename it to zip and unpack ...)
so after dowloading, zip de last commit files (being in the proper directory), change the file name extension to xpi
then at yout TB add-ons select the wheel icon and "install add-on from file"

**Comment by DavidBE-UK** [comment]
You don't need to change the extension from .zip to .xpi. Just 'install add-on from file' and select the .zip file and it will work.

**Comment by DavidBE-UK** [comment]
I just got the TB 102.3.3 update, so it's out.

**Comment by fdellwing** [comment]
Yes, 102.3.3 got added 7h ago to the mozillateam PPA.

**Comment by smmehta1822** [comment]
I got the update on Mac and manual installation of TBSync 4.2 fails with a message 'Not compatible with TB 102.3.3'.

**Comment by jobisoft** [comment]
> I got the update on Mac and manual installation of TBSync 4.2 fails with a message 'Not compatible with TB 102.3.3'.

Yes, working on a fix.

**Comment by ghost-from-the-past** [comment]
I updated on my MacOS to TB 102.3.3, without updating TbSyn + EAS-4. I continue to use what I donwloaded from the repository on october 6th. And it continues to work and synchronize with Exchange server ok (I just use calendar, no task not contacs)

**Comment by smmehta1822** [comment]
Thank you John and TBSync 4.1 and EAS-4-TB 4.1.4 continue to work fine.

**Comment by jobisoft** [comment]
> I got the update on Mac and manual installation of TBSync 4.2 fails with a message 'Not compatible with TB 102.3.3'.

Released TbSync v4.3 which fixes your issue. It has also been sent to ATN for review. Finalizing the EAS provider and submitting it later today as well.

**Comment by smmehta1822** [comment]
Thanks John. I guess will wait for official release unless there is a place to pull 4.3.

**Comment by jobisoft** [comment]
I have now released EAS v4.1.5.
https://github.com/jobisoft/EAS-4-TbSync/releases

This version also implements the hard limit to require Thunderbird 102.3.3 (it installs in 102.3.0 but shows a notification that 102.3.3 is required).

That the add-on install check is not able to use the 3rd level of the version number was new to me and I should have waited till the release of Thunderbird 102.4. Learning new things everyday.

EAS v4.1.5 is adjusting how categories are stored, so they will be picked up by my [CategoryManager](https://addons.thunderbird.net/addon/categorymanager/) add-on, which is being updated as well (no previews available yet).

If EAS v4.1.5 runs smoothly, I will officially release it to ATN on the weekend, so it could start to automatically distribute on Monday.

**Comment by jim-davenport** [comment]
Will this version include GAL?

**Comment by jobisoft** [comment]
> Will this version include GAL?

No, this will be covered in a separate issue after the initial release is done.

**Comment by jim-davenport** [comment]
Understood. For decision making purposes, would you guess days, weeks, or months, on GAL.  Thank you for your outstanding work.

**Comment by ghost-from-the-past** [comment]
I installed manually TbSyn 4.3 (commit from 18 hours ago) and EAS-4-TbSync (commit from 4 hours ago) on MacOS and TB 102.3.3 for me no problem, installation and calendar synch works ok
now I'll test on the Ubuntu 22.04.1

**Comment by smmehta1822** [comment]
Thank you John. I installed TBSync 4.3 and EAS 4.1.5 on TB 102.3.3 on Mac. And it's working well so far. Thank you again for your help and support.

**Comment by ghost-from-the-past** [comment]
for me, ok also on Ubuntu 22.04.01, manual install of the last commits

**Comment by michkot** [comment]
Tested for read-only sync of exchange, so far so good

**Comment by mmadela** [comment]
as far as I understand, the bug [1792399](https://bugzilla.mozilla.org/show_bug.cgi?id=1792399) is already fixed and released with 102.3.3 last week

**Comment by jobisoft** [comment]
And TbSync has been released after that, but review on ATN needs additional time. But even that has happened by now.

**Comment by smmehta1822** [comment]
Thank you John and the team.

**Comment by tigerjack** [comment]
Hey @jobisoft , thanks for your work!
What is the best way to handle the add-ons installed through the xpi files now? Uninstall TbSync and install the version provided on the add-ons website, while leaving the EAS add-on as it is?

**Comment by jobisoft** [comment]
You do not need to do anything. I always increased the version number on new releases, so you either have the most recent one already or get auto-updated via ATN (it is a myth that manually installed add-ons do not get updated).

**Comment by tigerjack** [comment]
> You do not need to do anything. I always increased the version number on new releases, so you either have the most recent one already or get auto-updated via ATN (it is a myth that manually installed add-ons do not get updated).
Thank you! I wasn't aware of that.

**Comment by DrYak** [comment]
Thanks for the update!

**Comment by Hornobster** [comment]
Thank you John, got the automatic update and all works fine.

**Comment by jonArzelus** [comment]
Hi, we got the automatic update and all seems to work fine, thanks for the work!

**Comment by manning-ncsa** [comment]
Thanks for all your hard work on this @jobisoft. I use it every day for work.

**Comment by rol1** [comment]
Thanks

**Comment by Ghostbird** [comment]
How hard are things going to break if I install this on Thunderbird 1.0.3.0? On Debian Thunderbird 1.0.3.3 is still in testing. I've removed the version check from the XPI and installed it by hand. For now I've set my calendar to read-only.

**Comment by smmehta1822** [comment]
Just upgraded to TB 102.4 and plugins seems to be working.

**Comment by rarunkarthik** [comment]
@jobisoft: Any update on GAL support?

**Comment by rarunkarthik** [comment]
@jobisoft any word on GAL support getting back in?

**Comment by Giermann** [comment]
> > Will this version include GAL?
> 
> No, this will be covered in a separate issue after the initial release is done.

Not only @rarunkarthik is waiting for GAL support to come back.
9 months ago, it seemed to be an issue that will be addressed just after the initial release. Is it that compicated with TB 102?
The add-on is becoming useless for me, because I cannot write to addresses, that have been created in the last 9 months or I have never written to in the time before...
